addons/mail/static/src/utils/test_utils.js

```diff
@@ -169,47 +174,90 @@ function nextAnimationFrame() {
+         // exists, update its default value according to the one in the
+         // registry; If it does not exist, add it to the model definition.
+        const fieldNamesToFieldToInsert = fieldsToInsertRegistry.category(modelName).getEntries();
+        for (const [fname, fieldToInsert] of fieldNamesToFieldToInsert) {
+            if (fname in fields) {
+                fields[fname].default = fieldToInsert.default;
+            } else {
+                fields[fname] = fieldToInsert;
+            }
+        }
+        // apply default values for date like fields if none was passed.
+        for (const fname in fields) {
+            const field = fields[fname];
+            if (['date', 'datetime'].includes(field.type) && !field.default) {
+                const defaultFieldValue =
+                    field.type === 'date'
+                        ? () => moment().format('YYYY-MM-DD')
+                        : () => moment.utc().format("YYYY-MM-DD HH:mm:ss");
+                field.default = defaultFieldValue;
+            }
+        }
+    }
+    modelDefinitions.set('res.fake', {
+        activity_ids: { string: "Activities", type: 'one2many', relation: 'mail.activity' },
+        email_cc: { type: 'char' },
+        partner_ids: { relation: 'res.partner', string: "Related partners", type: 'many2one', },
     });
+    return modelDefinitions;
+}
 
-    data.currentPartnerId = 3;
-    data['res.partner'].records.push({
-        display_name: "Your Company, Mitchell Admin",
-        id: data.currentPartnerId,
-        name: "Mitchell Admin",
-    });
-    data.currentUserId = 2;
-    data['res.users'].records.push({
-        display_name: "Your Company, Mitchell Admin",
-        id: data.currentUserId,
-        name: "Mitchell Admin",
-        partner_id: data.currentPartnerId,
-    });
+//------------------------------------------------------------------------------
+// Public: test lifecycle
+//------------------------------------------------------------------------------
 
-    data.publicPartnerId = 4;
-    data['res.partner'].records.push({
-        active: false,
-        display_name: "Public user",
-        id: data.publicPartnerId,
-    });
-    data.publicUserId = 3;
-    data['res.users'].records.push({
-        active: false,
-        display_name: "Public user",
-        id: data.publicUserId,
-        name: "Public user",
-        partner_id: data.publicPartnerId,
-    });
+async function beforeEach(self) {
+    const data = { ...TEST_USER_IDS };
+    const modelDefinitions = await modelDefinitionsPromise;
+    const recordsToInsertRegistry = registry.category('mail.model.definitions').category('recordsToInsert');
+    for (const [modelName, fields] of modelDefinitions) {
+        const records = [];
+        if (recordsToInsertRegistry.contains(modelName)) {
+            // avoid records corruption during the test.
+            records.push(...JSON.parse(JSON.stringify(recordsToInsertRegistry.get(modelName))));
```

Comment:
This is problematic, it should be done inside the mock server instead (or at least closer to it). A lot of lazy tests might rely on this behavior though, so probably to be taken care of later.